unstructured/partition/epub.py

```diff
@@ -29,13 +37,43 @@ def partition_epub(
         If True, the output will include page breaks if the filetype supports it
     metadata_last_modified
         The last modified date for the document.
-
     """
-    return convert_and_partition_html(
-        source_format="epub",
-        filename=filename,
-        file=file,
-        include_page_breaks=include_page_breaks,
-        metadata_filename=metadata_filename,
-        metadata_last_modified=metadata_last_modified,
-    )
+    exactly_one(filename=filename, file=file)
+    if filename is None:
+        filename = ""
+
+    if file is not None:
+        with tempfile.NamedTemporaryFile(delete=False) as tmp:
+            tmp.write(file.read())
+            filename = tmp.name
+        last_modification_date = get_last_modified_date_from_file(file)
+    else:
+        last_modification_date = get_last_modified_date(filename)
+
+    book = epub.read_epub(filename)
+    toc_items = list(book.toc)
+    elements = []
+
+    for toc_item in toc_items:
+        # Some toc items may be tuple
+        if isinstance(toc_item, tuple):
+            toc_item = toc_item[0]
+
+        href = toc_item.href.split("#")[0]
+        title = toc_item.title
```

Comment:
Do you know if the the TOC items corresponds to? Is it a chapter or something or does it depend on the book?

Reply:
I'm not 100% clear on that to be honest, but I know it includes more than just the "contents" section of a book. For example, the `winter-sports.epub` book we use for testing has the following toc items (plus the rest of the chapters):
'The Project Gutenberg eBook of Winter Sports in Switzerland, by E. F. Benson', 'WINTER SPORTS IN SWITZERLAND', 'CONTENTS', 'ILLUSTRATIONS', 'CHAPTER I THE SUN-SEEKER', etc.

And the final element in the toc items does include the text up to the end of the book (at least in this case--unsure how much variability there would be between different epub books)

Reply:
Gotcha, may be worth trying a couple of epubs from Project Gutenberg just to see what a few of them look like. What do you think about also added a section number to go along with the section name? Since there's no guarantee section names will be unique.

- https://www.gutenberg.org/

Reply:
Would we want to make partitioning via toc default and then have it partition via the existing way be an option users could pick via an argument?

Reply:
I don't think so. This way seems better since it doesn't require a system dependency (`pandoc`) and doesn't require the overhead of converting the doc. They can alway use `convert_and_partition_html` directly if they do want to convert with `pandoc` first and then partition.

Reply:
for this [book](https://www.gutenberg.org/cache/epub/6167/pg6167.html), I have the following toc items (and have verified that we're partitioning the beginning and ending of it):
(Pdb) titles
['CONTENTS', 'CHAPTER I COMMON SENSE FARMING 1. Wealth, Land and Labor. 2. Money. 3. Sources of Wealth. 4. The Farmer, a Producer, and Seller. 5. Business Methods Essential.', 'INTRODUCTION', 'CHAPTER I', 'CHAPTER II', 'CHAPTER III', 'CHAPTER IV', 'CHAPTER V', 'CHAPTER VI', 'CHAPTER VII', 'CHAPTER VIII', 'CHAPTER IX', 'CHAPTER X', 'CHAPTER XI', 'CHAPTER XII', 'CHAPTER XIII', 'CHAPTER XIV', 'CHAPTER XV', 'CHAPTER XVI', 'CHAPTER XVII', 'CHAPTER XVIII', 'CHAPTER XIX', 'CHAPTER XX', 'CHAPTER XXI', 'CHAPTER XXII', 'CHAPTER XXIII', 'CHAPTER XXIV', 'CHAPTER XXV', 'CHAPTER XXVI', 'CHAPTER XXVII', 'CHAPTER XXVIII', 'THE FULL PROJECT GUTENBERG LICENSE']
(Pdb) tuple_titles
['BUSINESS HINTS FOR MEN AND WOMEN']

and this [book](https://www.gutenberg.org/cache/epub/12511/pg12511-images.html)
(Pdb) titles
["The Project Gutenberg eBook of Blackwood's Edinburgh Magazine — Volume 53, No. 332, June, 1843", "BLACKWOOD'S EDINBURGH MAGAZINE", 'NO. CCCXXXII. JUNE, 1843. VOL. LIII.', 'CONTENTS.', "A PASSAGE IN THE LIFE OF A MAÎTRE-D'ARMES", "MR BAILEY'S REPLY TO AN ARTICLE IN BLACKWOOD'S MAGAZINE.", "THE ATTORNEY'S CLERK IN THE MONK'S HOOD.", 'COMMERCIAL POLICY—RUSSIA.', 'FOOTNOTES.', 'INDEX TO VOL. LIII.', 'END OF VOL. LIII.', 'THE FULL PROJECT GUTENBERG LICENSE']
(Pdb) tuple_titles
['MARSTON; OR, THE MEMOIRS OF A STATESMAN.', 'THE VIGIL OF VENUS.', 'CHAPTERS OF TURKISH HISTORY. RISE OF THE KIUPRILI FAMILY—SIEGE OF CANDIA.', 'AMMALÁT BEK.', 'THE VICAR OF WAKEFIELD.', 'IGNACIO GUERRA AND EL SANGRADOR;', "MEMORANDUMS OF A MONTH'S TOUR IN SICILY."]

Reply:
I am just finding, though, that the `toc` items are overlapping in some circumstances--the `html_content` for the second book I listed is the same for the first couple of items in the `toc`...

Reply:
Roger that - thanks!